yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -1173,3 +1186,160 @@ def run(self, info):
+        started = time.time()
+        status = {
+            'filename': self.info_dict.get('_filename'),
+            'status': self.running_status_value,
+            'total_bytes': total_filesize,
+            'elapsed': time.time() - started
+        }
+        # Parse ffmpeg_output
+        progress_pattern = re.compile(
+            r'(frame=\s*(?P<frame>\S+)\nfps=\s*(?P<fps>\S+)\nstream_0_0_q=\s*(?P<stream_0_0_q>\S+)\n)?bitrate=\s*(?P<bitrate>\S+)\ntotal_size=\s*(?P<total_size>\S+)\nout_time_us=\s*(?P<out_time_us>\S+)\nout_time_ms=\s*(?P<out_time_ms>\S+)\nout_time=\s*(?P<out_time>\S+)\ndup_frames=\s*(?P<dup_frames>\S+)\ndrop_frames=\s*(?P<drop_frames>\S+)\nspeed=\s*(?P<speed>\S+)\nprogress=\s*(?P<progress>\S+)')
+        retval = self.ffmpeg_proc.poll()
+        ffmpeg_stdout_buffer = ''
+        ffmpeg_stderr_buffer = ''
+        while retval is None:
+            ffmpeg_stdout = self.ffmpeg_proc.stdout.readline() if self.ffmpeg_proc.stdout else ''
+            ffmpeg_stderr_buffer = self.ffmpeg_proc.stderr.readline() if self.ffmpeg_proc.stderr else ''
+            if ffmpeg_stderr_buffer != '':
+                sys.stderr.write(ffmpeg_stderr_buffer)
+                ffmpeg_stderr_buffer = ''
+            if ffmpeg_stdout == '':
+                continue
+            ffmpeg_stdout_buffer += ffmpeg_stdout
+            ffmpeg_prog_infos = re.match(progress_pattern, ffmpeg_stdout_buffer)
+            if not ffmpeg_prog_infos:
+                continue
+            sys.stdout.write(ffmpeg_stdout_buffer)
+            eta_seconds = self.compute_eta(ffmpeg_prog_infos, duration_to_track)
+            bitrate_int = self.compute_bitrate(ffmpeg_prog_infos.group('bitrate'))
+            # Not using ffmpeg 'total_size' value as it's imprecise and give progress percentage over 100
+            out_time_second = int_or_none(ffmpeg_prog_infos.group('out_time_us')) // 1_000_000
+            try:
+                dl_bytes_int = int_or_none(out_time_second / duration_to_track * total_filesize)
+            except ZeroDivisionError:
+                dl_bytes_int = 0
+            status.update({
+                self.action_bytes_key: dl_bytes_int,
+                'speed': bitrate_int,
+                'eta': eta_seconds
+            })
+            self.hook_progress(status, self.info_dict)
+            ffmpeg_stdout_buffer = ''
+            status.update({'elapsed': time.time() - started})
+            retval = self.ffmpeg_proc.poll()
+            # Prevents ffmpeg_proc from hanging
+            if ffmpeg_prog_infos.group('progress') == 'end':
+                status.update({
+                    'status': 'finished',
+                    self.action_bytes_key: total_filesize
+                })
+                self.hook_progress(status, self.info_dict)
+                self.ffmpeg_proc.kill()
+                return ffmpeg_stdout_buffer, ffmpeg_stderr_buffer, 0
+
+        status.update({
+            'status': 'finished',
```

Comment:
`finished` hook should invoked only when the entire PP has run. A PP may have multiple ffmpeg runs. But it's fine to return finished here and we can handle it in the translation layer imo